material: don't initialize in-place (#1567)
cocos-creator/creator-docs: en/material-system/overview.md

The biggest difference between `Material` asset and `MaterialInstance` is: `MaterialInstance` is definitively attached to one `RenderableComponent` at the beginning of its life cyle, while `Material` has no such limit.

> For an existing material, you can easily modify the shader macros or pipeline states, with only the overrides you want to make:

Reviewer: ```suggestion
For an existing material, it is easy to modify the shader macros or pipeline states, with only the overrides necessary:
```

```ts
mat2.recompileShaders({
  USE_EMISSIVE: true // enable emissive on top of other macros
});
mat2.overridePipelineStates({
  rasterizerState: {
    cullMode: GFXCullMode.NONE // disable back face culling on top of other states
  }
});
```

> **Note**: these functions can only be invoked on `MaterialInstance`s, not `Material` asset itself.

Reviewer: ```suggestion
> **Note**: these functions can only be invoked on `MaterialInstance`s, not a `Material` asset itself.
```
